### PR TITLE
Delegate protocol to enable or disable the feature 

### PR DESCRIPTION
**Pull request by alimoeeny**
There are cases that you don't want the feature to be enabled. 
Examples: some users use the app while running around, and get annoyed by the prompt popping up all the time. 
To make it possible for the app to enable and disable the shake to bug report feature, I added a delegate protocol.

### CODE REVIEW

**Comment by dtrenz** [comment]
Great idea, Thanks!

File: Pod/Classes/BugShaker.swift [hunk 1]
```diff
@@ -8,12 +8,16 @@
...
+  func ShouldPresentReportPrompt() -> Bool
+}
 
 public class BugShaker {
-  
+
   struct Config {
     static var toRecipients: [String]?
     static var subject: String?
+    static var delegate: BugShakerDelegate?
```
Comment (dtrenz) [inline Pod/Classes/BugShaker.swift:18]: I tend to think of the `Config` struct as a collection of vars that will be used for the eventual email report. 

Could we move `delegate` to a `BugShaker` static class property?

File: Pod/Classes/BugShaker.swift [hunk 2]
```diff
@@ -8,12 +8,16 @@
 import UIKit
 import MessageUI
 
+public protocol BugShakerDelegate {
+  func ShouldPresentReportPrompt() -> Bool
```
Comment (dtrenz) [inline Pod/Classes/BugShaker.swift]: The general convention for Swift method names is camelCase. Could we change this to `shouldPresentReportPrompt()`?

File: Pod/Classes/BugShaker.swift [hunk 3]
```diff
@@ -45,7 +51,13 @@ extension UIViewController: MFMailComposeViewControllerDelegate {
   override public func motionEnded(motion: UIEventSubtype, withEvent event: UIEvent?) {
     if motion == .MotionShake {
       let cachedScreenshot = captureScreenshot()
-      
+
+      if let delegate = BugShaker.Config.delegate {
```
Comment (dtrenz) [inline Pod/Classes/BugShaker.swift]: Can we move this up to be the very first expression in the `motionEnded` method? That way we won't bother to capture a screenshot that we won't use.

File: Pod/Classes/BugShaker.swift [hunk 4]
```diff
@@ -45,7 +51,13 @@ extension UIViewController: MFMailComposeViewControllerDelegate {
...
+        if (!delegate.ShouldPresentReportPrompt()) {
```
Comment (dtrenz) [inline Pod/Classes/BugShaker.swift]: ^ Please remove the parens around the conditional statement.

**Comment by dtrenz** [comment]
A couple requests, beyond the line comments:
- Could you update the Example app to use the updated `BugShaker.configure()` method? (see: https://github.com/dtrenz/BugShaker/blob/develop/Example/BugShaker/AppDelegate.swift#L24)
- Would you mind updating the documentation of the `configure()` method example in the README?

**Comment by alimoeeny** [comment]
@dtrenz I added 3 new commits to address the changes. Let me know if I missed anything.

File: Example/BugShaker/AppDelegate.swift [hunk 5]
```diff
@@ -21,9 +21,17 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
 
-    BugShaker.configure(to: ["example@email.com"], subject: "Bug Report", body: "Hi Developers, I am reporting a bug where ... ")
+    BugShaker.configure(to: ["example@email.com"], subject: "Bug Report", body: "Hi Developers, I am reporting a bug where ... ", delegate: self)
     
     return true
   }
 
+    func shouldPresentReportPrompt() -> Bool {
+        // here you can read some settings from user defaults and decide if you want to have the bug report alert to show up or not
+        if NSUserDefaults.standardUserDefaults().stringForKey("CONFIG_ENABLE_BUGSHAKE") != "\(true)" {
```
Comment (dtrenz) [inline Example/BugShaker/AppDelegate.swift:31]: If you are testing for `true`, you could just use `boolForKey()`:

``` swift
return  NSUserDefaults.standardUserDefaults().boolForKey("CONFIG_ENABLE_BUGSHAKE")
```

Comment (alimoeeny) [inline Example/BugShaker/AppDelegate.swift:31]: I generally tend to write everything as string to `NSUserDefaults`. Have not tested with Swift, but at least in Obj C, `bookForKey` returns `false` both when you have written `false` to the userdefaults and when you have nothing there (which should return nil in my opinion). and there are some other oddities. 
But if you have strong feelings about this, please feel free to either check it as string or use `bookForKey`.

**Comment by dtrenz** [comment]
Please fix up the merge conflicts, and I will merge. Thanks, again!

**Comment by dtrenz** [comment]
Closing due to staleness. Deprecated by #14

**Pull request closed without merging**